crates/bgp-pkt/src/wire/serializer/bgp_ls.rs

```diff
@@ -0,0 +1,716 @@
+
+        match self {
+            BgpLsLinkDescriptorTlv::LinkLocalRemoteIdentifiers {
+                link_local_identifier,
+                link_remote_identifier,
+            } => {
+                writer.write_u32::<NetworkEndian>(*link_local_identifier)?;
+                writer.write_u32::<NetworkEndian>(*link_remote_identifier)?;
+            }
+            BgpLsLinkDescriptorTlv::IPv4InterfaceAddress(ipv4) => {
+                writer.write_u32::<NetworkEndian>((*ipv4).into())?
+            }
+            BgpLsLinkDescriptorTlv::IPv4NeighborAddress(ipv4) => {
+                writer.write_u32::<NetworkEndian>((*ipv4).into())?
+            }
+            BgpLsLinkDescriptorTlv::IPv6InterfaceAddress(ipv6) => {
+                writer.write_all(&ipv6.octets())?
+            }
+            BgpLsLinkDescriptorTlv::IPv6NeighborAddress(ipv6) => {
+                writer.write_all(&ipv6.octets())?
+            }
+            BgpLsLinkDescriptorTlv::MultiTopologyIdentifier(data) => data.write(writer)?,
+        };
+
+        Ok(())
+    }
+}
+
+impl WritablePdu<BgpLsWritingError> for BgpLsNodeDescriptorSubTlv {
+    const BASE_LENGTH: usize = 4; /* tlv type u16 + tlv length u16 */
+
+    fn len(&self) -> usize {
+        Self::BASE_LENGTH
+            + match self {
+                BgpLsNodeDescriptorSubTlv::AutonomousSystem(_) => 4,
+                BgpLsNodeDescriptorSubTlv::BgpLsIdentifier(_) => 4,
+                BgpLsNodeDescriptorSubTlv::OspfAreaId(_) => 4,
+                BgpLsNodeDescriptorSubTlv::IgpRouterId(inner) => inner.len(),
+                BgpLsNodeDescriptorSubTlv::BgpRouterIdentifier(_) => 4,
+                BgpLsNodeDescriptorSubTlv::MemberAsNumber(_) => 4,
+            }
+    }
+
+    fn write<T: Write>(&self, writer: &mut T) -> Result<(), BgpLsWritingError>
+    where
+        Self: Sized,
+    {
+        write_tlv_header(writer, self.get_type() as u16, self.len() as u16)?;
+        match self {
+            BgpLsNodeDescriptorSubTlv::AutonomousSystem(data) => {
+                writer.write_u32::<NetworkEndian>(*data)?
+            }
+            BgpLsNodeDescriptorSubTlv::BgpLsIdentifier(data) => {
+                writer.write_u32::<NetworkEndian>(*data)?
+            }
+            BgpLsNodeDescriptorSubTlv::OspfAreaId(data) => {
+                writer.write_u32::<NetworkEndian>(*data)?
+            }
+            BgpLsNodeDescriptorSubTlv::IgpRouterId(data) => writer.write_all(data)?,
+            BgpLsNodeDescriptorSubTlv::BgpRouterIdentifier(data) => {
+                writer.write_u32::<NetworkEndian>(*data)?
+            }
+            BgpLsNodeDescriptorSubTlv::MemberAsNumber(data) => {
+                writer.write_u32::<NetworkEndian>(*data)?
+            }
+        };
+
+        Ok(())
+    }
+}
+
+impl WritablePdu<BgpLsWritingError> for BgpLsNlriLink {
+    const BASE_LENGTH: usize = 1 /* protocol_id */ + 8 /* identifier */;
```

Comment:
I would use `//` comment to consistent with the rest of the code. No need of inline comments like this, I find them hard to read myself

Reply:
sorry, bad habit i picked up from FRRouting where they don't allow `//` comments...